docs/design/navigation-and-scale.md

```diff
@@ -0,0 +1,143 @@
+is:
+* introduction of use-case based navigation menu and views
+* replacement of free-form cards by tabularized resource lists
+* being explicit about Kubernetes resources displayed in the UI
+* introduction of action bar for performing all actions on resources
+
+## Navigation
+Navigation is realized by left hand side menu. The menu consists of use case
+driven categories that group Kubernetes resources. For example, there may be
+category called "Apps" which deals with Replication Controllers, Pods,
+Deployments, etc., and "Config" which deals with Secrets, Volumes,
+Namespaces, etc. The exact definition of categories will happen in a separate
+design.
+
+![Navigation](mockups/23-03-2016-scale-and-navigation/navigation.png)
+
+Menu categories may be expanded into raw resource subpages for expert users.
+
+![Menu extended](mockups/23-03-2016-scale-and-navigation/menu-extended.png)
+
+Menu follows standard responsiveness practises where it folds into icons and
+hamburger menu when space is limited.
+
+![Mobile navigation](mockups/23-03-2016-scale-and-navigation/mobile-navigation.png)
+
+## View templates
+
+This section shows concepts used as templates for all views.
+
+View which does not deal with a resource or a resource list, e.g., landing
+page which shows an overview of a cluster. The page consists of, potentially
+dismissable, blocks that display data.
+![Landing page](mockups/23-03-2016-scale-and-navigation/landing-page.png)
+
+Category view which shows lists of resources. In this mockup it is
+(not defined yet) "Apps" category which was selected through the
+navigation menu.
+![Category page template](mockups/23-03-2016-scale-and-navigation/category-page-template.png)
+
+Action bar replaces floating action button. The reason for this change
+is the fact that floating action buttons do not work well on desktop
+form factors, which are used by vast majority of our users. On the other
+hand, action bars work well on any form factors (desktop, mobile)
+and can scale for different actions on different resources.
+![Action bar](mockups/23-03-2016-scale-and-navigation/action-bar.png)
+
+Details of the resource list block:
+![Resource list block](mockups/23-03-2016-scale-and-navigation/resource-list-block.png)
+
+While the concept of cards used for displaying resources has some nice
+characteristics (like information density) it has been a constant source of
+problems. Users and developers complained about many issues: how do you sort
+cards? how do you change their order? how do you paginate them? how do you
+compare resources against each other? how do you distinct different resources
+in a card stream? how do you do bulk operations?
+
+We have identified that in order to solve these problems the cards can evolve
+into tabularized list items which can show exactly the same information as
+cards. This solution has many advantages, some of which are: pagination,
+ordering, bulk actions or visual comparison. Following mockup outlines
```

Comment:
checkboxes would be the typical approach, it provides a clear indicator that the row is selectable, and provides a browser-native accessible selector for keyboard+screenreader users